AsyncDisplayKit/ASCellNode.mm

```diff
@@ -204,6 +204,26 @@ - (void)__setHighlightedFromUIKit:(BOOL)highlighted;
   }
 }
 
+- (NSIndexPath *)indexPath
+{
+  ASDisplayNodeAssertMainThread();
+
+  UIScrollView *scrollView = self.scrollView;
+
+  ASDisplayNode *owningNode = scrollView.asyncdisplaykit_node;
+  if ([owningNode isKindOfClass:[ASCollectionNode class]]) {
```

Comment:
The isKindOfClass: calls are pretty heavy ... can we move this logic in setScrollView: and cache via a flags property? Look into `ASCollectionView` / `ASTableView` how we check if delegate methods are implemented.